optaplanner-examples/src/test/java/org/optaplanner/examples/common/experimental/ConsecutiveSetTreeTest.java

```diff
@@ -286,6 +287,46 @@ public void testRandomSequences() {
         }
     }
 
+    @Test
+    public void testRandomSequencesWithDuplicates() {
```

Comment:
I think that tests like these should be parameterized. This is not one test - this is a 1000 different tests, each with 2 assertions.